tests/unit/src/test/scala/tests/DebugDiscoverySuite.scala

```diff
@@ -577,4 +577,591 @@ class DebugDiscoverySuite
+    for {
+      _ <- initialize(
+        s"""/metals.json
+           |{
+           |  "a": {
+           |    "libraryDependencies":["org.scalatest::scalatest:3.2.16"]
+           |  }
+           |}
+           |/${fooPath}
+           |package a
+           |
+           |class Foo extends org.scalatest.funsuite.AnyFunSuite {
+           |  test("first") {
+           |    print("first test")
+           |  }
+           |  test("second") {
+           |    print("second test") // <- Cursor in this line
+           |  }
+           |}
+           |""".stripMargin
+      )
+      _ <- server.didOpen(fooPath)
+      _ <- server.didSave(fooPath)
+      _ <- server.waitFor(TimeUnit.SECONDS.toMillis(10))
+      debugger <- server.startDebuggingUnresolved(
+        new DebugDiscoveryParams(
+          server.toPath(fooPath).toURI.toString,
+          "runClosest",
+          position = new org.eclipse.lsp4j.Position(7, 5),
+        ).toJson
+      )
+      _ <- debugger.initialize
+      _ <- debugger.launch
+      _ <- debugger.configurationDone
+      _ <- debugger.shutdown
+      output <- debugger.allOutput
+    } yield assert(
+      output.contains("second test") &&
+        !output.contains("first test") &&
+        output.contains("1 tests, 1 passed")
+    )
+  }
+
+  test("run-closest-suite") {
+    for {
+      _ <- initialize(
+        s"""/metals.json
+           |{
+           |  "a": {
+           |    "libraryDependencies":["org.scalatest::scalatest:3.2.16"]
+           |  }
+           |}
+           |/${fooPath}
+           |package a
+           |
+           |class Foo extends org.scalatest.funsuite.AnyFunSuite { // <- Cursor in this line
+           |  test("first") {
+           |    print("first test")
+           |  }
+           |  test("second") {
+           |    print("second test")
+           |  }
+           |}
+           |""".stripMargin
+      )
+      _ <- server.didOpen(fooPath)
+      _ <- server.didSave(fooPath)
+      _ <- server.waitFor(TimeUnit.SECONDS.toMillis(10))
+      debugger <- server.startDebuggingUnresolved(
+        new DebugDiscoveryParams(
+          server.toPath(fooPath).toURI.toString,
+          "runClosest",
+          position = new org.eclipse.lsp4j.Position(2, 4),
+        ).toJson
+      )
+      _ <- debugger.initialize
+      _ <- debugger.launch
+      _ <- debugger.configurationDone
+      _ <- debugger.shutdown
+      output <- debugger.allOutput
+    } yield assert(
+      output.contains("first test") &&
+        output.contains("second test") &&
+        output.contains("2 tests, 2 passed")
+    )
+  }
+
+  test("run-closest-multiple-suites") {
+    for {
+      _ <- initialize(
+        s"""/metals.json
+           |{
+           |  "a": {
+           |    "libraryDependencies":["org.scalatest::scalatest:3.2.16"]
+           |  }
+           |}
+           |/${fooPath}
+           |package a
+           |
+           |class Foo extends org.scalatest.funsuite.AnyFunSuite {
```

Comment:
```suggestion
           |class BarFoo extends org.scalatest.funsuite.AnyFunSuite {
```
would test the issue (I think might be happening) with contains that you use in DebugDiscovery. Unless that's not a problem, my head compiler seems to think so 😅